auraed/src/runtime/executable_name.rs

```diff
@@ -0,0 +1,61 @@
+use std::fmt::{Display, Formatter};
+use std::ops::Deref;
+use validation::{ValidatedField, ValidationError};
+
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub(crate) struct ExecutableName(String);
+
+impl ExecutableName {
+    pub fn validate_for_creation(
+        input: Option<String>,
```

Comment:
why would we ever pass None to validation?

Reply:
Obviously we shouldn't, or validation would fail, but the function doesn't know that you will always have a `String`. Essentially, it is validating that you even have a value. Given the way the api/protos are currently, we wouldn't have that scenario.

However, I feel there are benefits to keeping a consistent signature style for the `validate_for_creation` functions. I probably should have made it a function on the `ValidatedField` trait, with a default implementation. I'll do that.